web/src/settings_exports.ts

```diff
@@ -203,16 +318,62 @@ function show_start_export_modal(): void {
 export function set_up(): void {
     meta.loaded = true;
 
+    const toggler = components.toggle({
+        child_wants_focus: true,
+        values: [
+            {label: $t({defaultMessage: "Data exports"}), key: "data-exports"},
+            {label: $t({defaultMessage: "Export permissions"}), key: "export-permissions"},
+        ],
+        callback(_name, key) {
+            $(".export_section").hide();
+            $(`[data-export-section="${CSS.escape(key)}"]`).show();
+        },
+    });
+
+    toggler.get().prependTo($("#data-exports .tab-container"));
+    toggler.goto("data-exports");
+
+    // Do an initial population of the 'Export permissions' table
     void channel.get({
         url: "/json/export/realm/consents",
         success(raw_data) {
             const data = z
                 .object({export_consents: z.array(export_consent_schema)})
                 .parse(raw_data);
-            total_users_count = data.export_consents.length;
-            users_consented_for_export_count = data.export_consents.filter(
+            export_consents = data.export_consents;
+
+            // Apply queued_export_consents on top of the received response.
+            // This loop has O(n^2) time-complexity, but 'queued_export_consents'
+            // would be of a small length, making it effectively O(n).
```

Comment:
Is this assumption acceptable ?

Reply:
It's basically fine from a practical performance standpoint, but also you should just make the data structure a `map` from user IDs to their consent status, and then this code will get way way simpler and it'll also naturally eliminate the performance issue. Almost always, when the server sends a list to the client, the client's first thing to do will be to convert into a `map` from the primary key to the data, since while lists are simpler as a wire protocol, maps are cheaper to update.

Reply:
Updated code to use map.